src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -80,8 +98,39 @@ def __init__(self, start: int, end: int):
         )
         self._condition = threading.Condition()
         self._no_remote_peer = 0
+        self._metadata_keys = [key.lower() for key in metadata_keys]
+        self._include_all_metadata = "*" in [
+            key.strip() for key in metadata_keys
+        ]
+        self._metadata_by_peer = collections.defaultdict(
+            messages_pb2.LoadBalancerStatsResponse.MetadataByPeer
+        )
 
-    def on_rpc_complete(self, request_id: int, peer: str, method: str) -> None:
+    def _add_metadata(
+        self,
+        rpc_metadata: messages_pb2.LoadBalancerStatsResponse.RpcMetadata,
+        metadata: MetadataType,
+        type: messages_pb2.LoadBalancerStatsResponse.MetadataType,
+    ) -> None:
+        for key, value in metadata:
+            if self._include_all_metadata or key.lower() in self._metadata_keys:
```

Comment:
You're stripping the key when searching for `*`, but not when adding they a key. I wonder if we should always strip, or don't strip at all.

Reply:
I'm following CPP client: https://github.com/grpc/grpc/blob/d4f1f8543e38af756dbecd2f4dd5816d36866efd/test/cpp/interop/xds_stats_watcher.cc#L74-L77

No matter what we decide, I think we should have a consistent behavior from those two clients.